### PR TITLE
KUIF-61 Add Docs to KUIF project

### PR DESCRIPTION
**Pull request by mariokleen**
*Changes:*
- Installed compodoc to generate documentation 
- Modified gitignore to discard docmentation folder
- updated Dockerfile to generate and serve the documentation
- tsconfig outDir set to root in order to include projects folder for
documentation purposes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,18 +3,20 @@
...
     "build:all": "yarn build:core && yarn build:components",
-    "test": "ng test",
-    "lint": "ng lint",
+    "build:elements": "rimraf elements && rimraf dist/kui-elements && ng build kui-elements --prod --output-hashing none --build-optimizer && node build-elements-script.js",
+    "build:components": "ng build components",
+    "build:core": "ng build core",
+    "build:prod": "yarn build:all && ng build --aot=false --progress=false --prod --build-optimizer=false",
+    "clean": "rimraf dist",
+    "compodoc": "compodoc src -s",
+    "compodoc:build": "compodoc -p ./tsconfig.json --hideGenerator --silent",
```
Comment (sebaskleeen) [inline package.json]: What do you think about naming the scripts `build:docs` and `docs`. to align them with the other yarn scripts we have. Or do you think is better to name them differently so they don't get confused with another project?

Comment (mariokleen) [reply package.json]: sounds good, like this?

docs:  "compodoc src -s"
build:docs: "compodoc -p ./tsconfig.json --hideGenerator --silent",

Comment (sebaskleeen) [reply package.json]: yep exactly 👍

Comment (mariokleen) [reply package.json]: Regarding the README I agree, I think it's just a copy and paste from an old project, we'll need to update it

Comment (sebaskleeen) [reply package.json]: ok we can do that as part of another ticket related to docker, not documentation then :)

**Pull request merged**